src/main/java/net/minestom/server/network/packet/client/status/StatusRequestPacket.java

```diff
@@ -27,15 +30,22 @@ public void process(@NotNull PlayerConnection connection) {
         responseData.setDescription(Component.text("Minestom Server"));
         responseData.setFavicon("");
 
-        if (consumer != null)
+        if (consumer != null) {
             consumer.accept(connection, responseData);
+        }
 
         // Call event
         ServerListPingEvent statusRequestEvent = new ServerListPingEvent(responseData, connection);
         MinecraftServer.getGlobalEventHandler().callCancellableEvent(ServerListPingEvent.class, statusRequestEvent,
                 () -> {
-                    ResponsePacket responsePacket = new ResponsePacket();
-                    responsePacket.jsonResponse = responseData.build().toString();
+                    final ResponsePacket responsePacket = new ResponsePacket();
+
+                    // check if we need to use a legacy response
+                    if (connection.getProtocolVersion() >= 713) {
```

Comment:
Would probably be a bit clearer if this was a constant, e.g. `LEGACY_PROTOCOL_VERSION`. Also, due to there being two sets of versioning (versioning was reset for the Netty rewrite), so this could be confused with the new versions, couldn't it?

Reply:
Nevermind me, looks like this has been solved